WIP ReadMe (#5)
pivotal-cf/docs-build-service: README.md

# Pivotal Build Service

> Build Service is a source code to container (S2I) solution that utilizes the OSS [Cloud Native Buildpacks project](http://buildpacks.io) to execute consistent builds that adhere to [modern container standards](

Reviewer: ```suggestion
Build Service is a source code to container (S2I) solution that utilizes the OSS [Cloud Native Buildpacks project](http://buildpacks.io) to execute reproducible builds that adhere to [modern container standards](
```

https://github.com/opencontainers/image-spec/blob/master/spec.md).  Additionally, Build Service employs declarative logic to automate image re-builds.

> Understanding the differences between declarative and imperative programming models as they relate to producing container images from source code is key to deriving the most value possible from Build Service.  Pivotal Application Service, based on Cloud Foundry, currently follows an imperative model.  The user instructs the platform to perform various tasks through the `cf push` model.  Similarly, the [Cloud Native Buildpacks project](http://buildpacks.io), follows an imperative model.  Users perform commands like `build` and `rebase` using the `pack` CLI to trigger one-time execution of these commands.

Reviewer 2: 💯 I really like this simple yet critical piece of information.

The `pb` (Pivotal Build) CLI doesn’t have commands like `push`, `build`, or `rebase`.  Instead, users utilize `pb` to declare and modify two configurations.  

> 1) The “Image” config:  Users describe their app through several inputs, and build service will deliver new images that resolve discrepancies between how the app is described and the latest version of the image. When the user requests the latest version of an application dependency, and that dependency becomes available, a new image is automatically built to replace the old image.

Reviewer: ```suggestion
1) The “Image” config:  Users describe their app through several inputs, and build service will deliver new images that resolve discrepancies between how the app is described and the latest version of the image. When the user requests the latest version of an application dependency, and that dependency becomes available, a new image is automatically built.
```

> 1) The “Image” config:  Users describe their app through several inputs, and build service will deliver new images that resolve discrepancies between how the app is described and the latest version of the image. When the user requests the latest version of an application dependency, and that dependency becomes available, a new image is automatically built to replace the old image.

Reviewer: ```suggestion
1) The “Image” config:  Users describe their app through several inputs, and build service will deliver new images that resolve discrepancies between how the app is described and the latest version of the image. When a new version of an application dependency becomes available, an image is automatically built with the new version of the dependency.
```

2) The “Team” config:   An extension to build service that provides access control to image configs using UAA.

> Through utilizing these configs, platform engineers and application developers can more easily and securly enjoy the benefits of containerized software development at enterprise scale.  

Reviewer: ```suggestion
These configurations help platform engineers and application developers enjoy the benefits of containerized software development at enterprise scale more easily and securely.  
```
